index.js

```diff
@@ -12,32 +23,66 @@ const eth = new Eth(provider);
 
 var express = require('express'),
     cors = require('cors'),
-    app = express();
+    app = express(),
+    bodyParser = require('body-parser'),
+    jsonParser = bodyParser.json();
 
 app.use(cors());
 
-app.get('/address/:address', function(req, res, next){
-  eth.getTransactionCount(address, (err, nonce) => {
-    eth.sendTransaction({
-      from: address,
-      to: req.params.address,
-      gas: 100000,
-      value: (parseFloat(process.env.AMOUNT) || 1.5) * 1e18,
-      data: '0xde5f72fd', // sha3('faucet()')
-      nonce,
-    }, (err, txID) => {
-      if (err) {
-        console.log('Request failed', err)
-        return res.status(500).json(err)
+// Store issue ids and their bounty addresses
+var issueData = {};
+
+// Receive a POST request at the address specified by an env. var.
+app.post('/address/:address', jsonParser, function(req, res, next){
+  if (!req.body)
+    return res.sendStatus(400);
+  var commentBody = req.body.comment.body;
+  var issueId = req.body.issue.id;
+  var namePosition = commentBody.search("@" + name);
+  // Store toAddress from commiteth
+  if (namePosition == -1) {
+    issueData[issueId] = {"toAddress": commentBody.substring(commentBody.search("Contract address:") + 18, commentBody.search("Contract address:") + 60)}
+    console.log(issueData);
+    return res.status(204);
+  }
+  else {
+    var postNameWords = commentBody.substring(namePosition + 1 + name.length + 1).trim().split(' ');
+    var amount = 0;
+    if (postNameWords.length > 0) {
+      if(postNameWords[0] == "standard") {
+        amount = process.env.STANDARD_BOUNTY;
       }
       else {
-        console.log('Successful request:', txID)
-        res.json({ txID })
+        amount = parseFloat(postNameWords[0]);
       }
+    }
+    console.log("Trying to give " + amount + " ETH to " + issueData[issueId].toAddress + " for issue " + issueId);
+    issueData[issueId].amount = amount;
+
+    // Conduct the transaction
+    eth.getTransactionCount(address, (err, nonce) => {
+      eth.sendTransaction({
+        from: address, // Specified in webhook, secret
```

Comment:
Oh waitamin, what's the fromAddress of the eth. transaction supposed to be? Is it the same as the env variable? Then everybody who can access the blockchain (or the receiver of transactions), would know this and can jeopardize everything.